sklearn/utils/estimator_checks.py

```diff
@@ -1460,6 +1474,65 @@ def check_estimators_dtypes(name, estimator_orig, strict_mode=True):
                 getattr(estimator, method)(X_train)
 
 
+def check_transformer_preserve_dtypes(
+    name, transformer_orig, strict_mode=True
+):
+    # check that dtype are preserved meaning if input X is of some dtype
+    # X_transformed should be from the same dtype.
+    X, y = make_blobs(
+        n_samples=30,
+        centers=[[0, 0, 0], [1, 1, 1]],
+        random_state=0,
+        cluster_std=0.1,
+    )
+    X = StandardScaler().fit_transform(X)
+    X -= X.min()
+    X = _pairwise_estimator_convert_X(X, transformer_orig)
+    assert X.dtype == np.float64
+
+    def _fit_and_transform(transformer, X, y):
+        transformer = clone(transformer)
+        set_random_state(transformer)
+        if hasattr(transformer, 'fit_transform'):
+            X_trans = transformer.fit_transform(X, y)
+        else:
+            X_trans = transformer.fit(X, y).transform(X)
```

Comment:
Are there a lot of methods that have fit and transform but not fit_transform? Shouldn't they inherit from `TransformerMixin` then?